src/app/features/lp-form/reducer.ts

```diff
@@ -710,29 +724,29 @@ export const slice = createSlice({
         updateLeverageOptionsAfterGetInfoPostLp(state);
         validateUserInputAndUpdateSubmitButton(state);
       })
-      .addCase(setSignerAndPositionForAMMThunk.pending, (state) => {
-        state.position.value = null;
-        state.position.status = 'pending';
+      .addCase(setSignerAndPositionsForAMMThunk.pending, (state) => {
+        state.positions.value = null;
+        state.positions.status = 'pending';
         if (!state.amm) {
           return;
         }
         state.amm.signer = null;
       })
-      .addCase(setSignerAndPositionForAMMThunk.rejected, (state) => {
-        state.position.value = null;
-        state.position.status = 'error';
+      .addCase(setSignerAndPositionsForAMMThunk.rejected, (state) => {
+        state.positions.value = null;
+        state.positions.status = 'error';
         if (!state.amm) {
           return;
         }
         state.amm.signer = null;
       })
-      .addCase(setSignerAndPositionForAMMThunk.fulfilled, (state, { payload }) => {
-        state.position.value = (payload as SetSignerAndPositionForAMMThunkSuccess).position;
-        state.position.status = 'success';
+      .addCase(setSignerAndPositionsForAMMThunk.fulfilled, (state, { payload }) => {
+        state.positions.value = (payload as SetSignerAndPositionsForAMMThunkSuccess).positions;
+        state.positions.status = 'success';
         if (!state.amm) {
```

Comment:
if would move this on start of function. Basically if no amm, return early and don't do anything. It is not possible to have positions without AMM